### PR TITLE
Document G8sControlPlane CRD

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/8804

- Adds YAML CRD and CR files generated by unit tests
- Adds property descriptions to the schema

### CODE REVIEW

File: docs/crd/infrastructure.giantswarm.io_g8scontrolplane.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          This resource represents an abstract control plane (Kubernetes master node and Etcd server)
+          of a tenant cluster in a Giant Swarm installation. It is reconciled by cluster-operator.
+          For implementation details it references a provider specific resource.
+        properties:
+          spec:
+            properties:
+              infrastructureRef:
+                description: |
+                  Reference to an [AWSControlPlane](https://docs.giantswarm.io/reference/cp-k8s-api/awscontrolplanes.infrastructure.giantswarm.io/)
+                  resource defining provider specific details for the c
```
Comment (xh3b4sd) [inline docs/crd/infrastructure.giantswarm.io_g8scontrolplane.yaml:34]: For the what? There got probably something lost.

File: pkg/apis/infrastructure/v1alpha2/g8s_control_plane_types.go [hunk 2]
```diff
@@ -79,6 +89,18 @@ func NewG8sControlPlaneTypeMeta() metav1.TypeMeta {
 	}
 }
 
+// NewG8sControlPlaneCR returns a Cluster Custom Resource.
```
Comment (xh3b4sd) [inline pkg/apis/infrastructure/v1alpha2/g8s_control_plane_types.go:92]: I am not sure "Cluster Custom Resource" is correct in this case. It is also that the value of the type returned is mostly empty. Do we really need that function?

Comment (marians) [inline pkg/apis/infrastructure/v1alpha2/g8s_control_plane_types.go:92]:
```suggestion
// NewG8sControlPlaneCR returns a G8sControlPlane Custom Resource.
```

Comment (marians) [reply pkg/apis/infrastructure/v1alpha2/g8s_control_plane_types.go]: Fixed

**Pull request merged**